server/src/main/java/org/opensearch/action/bulk/TransportBulkAction.java

```diff
@@ -644,15 +652,15 @@ public void onFailure(Exception e) {
                         // create failures for all relevant requests
                         for (BulkItemRequest request : requests) {
                             final String indexName = concreteIndices.getConcreteIndex(request.index()).getName();
-                            DocWriteRequest<?> docWriteRequest = request.request();
-                            responses.set(
+                            final DocWriteRequest<?> docWriteRequest = request.request();
+                            final BulkItemResponse bulkItemResponse = new BulkItemResponse(
                                 request.id(),
-                                new BulkItemResponse(
-                                    request.id(),
-                                    docWriteRequest.opType(),
-                                    new BulkItemResponse.Failure(indexName, docWriteRequest.id(), e)
-                                )
+                                docWriteRequest.opType(),
+                                new BulkItemResponse.Failure(indexName, docWriteRequest.id(), e)
```

Comment:
We need to account for operations through TransportUpdateAction as well since they will not go through TransportBulkAction. Can you handle incrementing the status counter there as well?

Reply:
Nice, catch! Thanks.

Reply:
Turns out [TransportUpdateAction](https://github.com/opensearch-project/OpenSearch/blob/2.9/server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java) delegates to single item bulk request. So, the flow eventually hit the same path.

Reply:
We may miss no-ops 

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java#L333